src/tests/CMakeLists.txt

```diff
@@ -53,16 +53,25 @@ find_package(Boost 1.57.0 REQUIRED COMPONENTS unit_test_framework)
   list(APPEND test_cases test_jobtopology)
   list(APPEND test_cases test_dipoledipoleinteraction)
   list(APPEND test_cases test_populationanalysis)
-list(APPEND test_cases test_gaussian)
-list(APPEND test_cases test_orca)
-list(APPEND test_cases test_nwchem)
-list(APPEND test_cases test_dftengine)
-list(APPEND test_cases test_bsecoupling)
-list(APPEND test_cases test_rate_engine)
+  list(APPEND test_cases test_gaussian)
+  list(APPEND test_cases test_orca)
+  list(APPEND test_cases test_nwchem)
+  list(APPEND test_cases test_dftengine)
+  list(APPEND test_cases test_bsecoupling)
+  list(APPEND test_cases test_rate_engine)
+  if(USE_GPU)
+    list(APPEND test_cases test_eigencuda)
+  endif()
 
   foreach(PROG ${test_cases} )
     add_executable(unit_${PROG} ${PROG}.cc)
-    target_link_libraries(unit_${PROG} votca_xtp ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
+    if(USE_GPU)
```

Comment:
You can leave the common case outside of the `if(USE_GPU)` as `target_include_directories()` and `target_include_directories()` always add to the lists.